src/main/java/io/nats/client/ConsumerInfo.java

```diff
@@ -0,0 +1,203 @@
+// Copyright 2020 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io.nats.client;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.nats.client.impl.JsonUtils;
+import io.nats.client.impl.JsonUtils.FieldType;
+
+// TODO Add properties
+
+/**
+ * The ConsumerInfo class returns information about a jetstream consumer.
+ */
+public class ConsumerInfo {
+
+    /**
+     * This class holds the sequence numbers for a consumer and 
+     * stream.
+     */
+    public class SequencePair {

```

Comment:
This is sort nitty.I don't think this needs to be public, and it's not strictly an inner class in it's usage, more like a companion.
```
private static class SequencePair {
```